quipucords/tests/integration/test_network_scan.py

```diff
@@ -0,0 +1,325 @@
+        "connection_port",
+        "connection_timestamp",
+        "connection_uuid",
+        "cpu_bogomips",
+        "cpu_core_count",
+        "cpu_core_per_socket",
+        "cpu_count",
+        "cpu_cpu_family",
+        "cpu_hyperthreading",
+        "cpu_model_name",
+        "cpu_model_ver",
+        "cpu_siblings",
+        "cpu_socket_count",
+        "cpu_vendor_id",
+        "date_date",
+        "date_machine_id",
+        "decision_central_candidates",
+        "decision_central_candidates_eap",
+        "eap5_home_candidates",
+        "eap_home_candidates",
+        "etc_issue",
+        "etc_machine_id",
+        "etc_release_name",
+        "etc_release_release",
+        "etc_release_version",
+        "fuse_activemq_version",
+        "fuse_camel_version",
+        "fuse_cxf_version",
+        "jboss_brms_business_central_candidates",
+        "jboss_brms_decision_central_candidates",
+        "jboss_brms_kie_server_candidates",
+        "jboss_eap_common_files",
+        "jboss_eap_id_jboss",
+        "jboss_eap_packages",
+        "jboss_processes",
+        "jws_has_cert",
+        "jws_home_candidates",
+        "jws_installed_with_rpm",
+        "jws_version",
+        "karaf_homes",
+        "kie_search_candidates",
+        "kie_server_candidates",
+        "kie_server_candidates_eap",
+        "redhat_packages_certs",
+        "redhat_packages_gpg_is_redhat",
+        "redhat_packages_gpg_last_built",
+        "redhat_packages_gpg_last_installed",
+        "redhat_packages_gpg_num_installed_packages",
+        "redhat_packages_gpg_num_rh_packages",
+        "system_purpose_json",
+        "uname_all",
+        "uname_hardware_platform",
+        "uname_hostname",
+        "uname_kernel",
+        "uname_os",
+        "uname_processor",
+        "user_has_sudo",
+        "virt_type",
+        "virt_virt",
+    }
+
+
+@pytest.fixture
+def expected_middleware_names():
+    """Middleware names."""
+    return {"JBoss BRMS", "JBoss EAP", "JBoss Web Server", "JBoss Fuse"}
+
+
+@pytest.fixture
+def fingerprint_fact_map():
+    """Map fingerprint to raw fact name."""
+    return {
+        "name": "uname_hostname",
+        "architecture": "uname_processor",
+        "redhat_package_count": "redhat_packages_gpg_num_rh_packages",
+        "redhat_certs": "redhat_packages_certs",
+        "is_redhat": "redhat_packages_gpg_is_redhat",
+        "etc_machine_id": "etc_machine_id",
+        "os_name": "etc_release_name",
+        "os_version": "etc_release_version",
+        "os_release": "etc_release_release",
+        "cpu_count": "cpu_count",
+        "cpu_socket_count": "cpu_socket_count",
+        "cpu_core_count": "cpu_core_count",
+        "cpu_core_per_socket": "cpu_core_per_socket",
+        "cpu_hyperthreading": "cpu_hyperthreading",
+        "system_last_checkin_date": "connection_timestamp",
+        "infrastructure_type": "virt_what_type/virt_type",
+        "system_creation_date": "date_machine_id",
+    }
+
```

Comment:
You wanted to cover all facts generated by the fingerprinter step or just a few of the most common ones ?

Reply:
only the ones that are being collected.
This will require a refactor when all facts are collected